src/test/java/org/broadinstitute/hellbender/utils/io/IOUtilsUnitTest.java

```diff
@@ -41,6 +44,212 @@ public void tearDown() {
+                        if ( fileAttributes.isRegularFile() && !fileAttributes.isSymbolicLink()) {
+                            try {
+                                final byte[] actualContents   = Files.readAllBytes(actualPath);
+                                final byte[] expectedContents = Files.readAllBytes(expectedPath);
+
+                                Assert.assertEquals(actualContents, expectedContents);
+                            }
+                            catch (final IOException ex) {
+                                throw new GATKException("Could not verify contents of files: " + actualPath.toUri() + " AND " + expectedPath.toUri(), ex);
+                            }
+                        }
+
+                        return true;
+                    }
+            );
+        }
+        catch (final IOException ex) {
+            throw new GATKException("Could not verify identical contents of test directories!");
+        }
+    }
+
+    @DataProvider
+    private Object[][] provideForTestExtractTarGz() {
+        return new Object[][] {
+                { null },
+                { createTempDir("provideForTestExtractTarGz_Case2").toPath() },
+        };
+    }
+
+    @Test(dataProvider = "provideForTestExtractTarGz")
+    public void testExtractTarGz(final Path destDir) {
+
+        final Path outputDataSourcesPath;
+
+        // Prepare our output directory:
+        if ( destDir == null ) {
+            final File tmpDir = createTempDir("IOUtilsUnitTest_testExtractTarGz");
+            tmpDir.deleteOnExit();
+            final Path tmpDirPath            = tmpDir.toPath();
+            outputDataSourcesPath = tmpDirPath.resolve(IOUtils.getPath(FuncotatorTestConstants.DUMMY_DATA_SOURCES_TAR_GZ).getFileName());
+
+            // Copy our data sources to the destination folder:
+            try {
+                Files.copy(IOUtils.getPath(FuncotatorTestConstants.DUMMY_DATA_SOURCES_TAR_GZ), outputDataSourcesPath);
+            }
+            catch (final IOException ex) {
+                throw new GATKException("Could not copy files for testing!", ex);
+            }
+
+            // Extract the files:
+            IOUtils.extractTarGz( outputDataSourcesPath );
+        }
+        else {
+            // Extract the files:
+            IOUtils.extractTarGz( IOUtils.getPath(FuncotatorTestConstants.DUMMY_DATA_SOURCES_TAR_GZ), destDir );
+
+            outputDataSourcesPath = destDir.resolve(IOUtils.getPath(FuncotatorTestConstants.DUMMY_DATA_SOURCES_TAR_GZ).getFileName());
+        }
+
+        // Get the base expected data sources output path:
+        final Path baseExpectedOutputDataSourcesPath = IOUtils.getPath(FuncotatorTestConstants.DUMMY_DATA_SOURCES_FOLDER);
+
+        // Get the base file path for the output:
+        final Path baseActualOutputDataSourcesPath = outputDataSourcesPath.getParent().resolve(
+                baseExpectedOutputDataSourcesPath.getFileName()
```

Comment:
Isn't `outputDataSourcesPath.getParent().resolve(baseExpectedOutputDataSourcesPath.getFileName())` the same as just `outputDataSourcesPath`?

Reply:
No - `outputDataSourcesPath` is the path to the `tar.gz` file whereas `baseActualOutputDataSourcesPath` is the path to the extracted folder.